dns/dnscrypt-proxy2/src/opnsense/mvc/app/models/OPNsense/Dnscryptproxy/General.xml

```diff
@@ -0,0 +1,133 @@
+            <MinimumValue>1</MinimumValue>
+            <MaximumValue>10000</MaximumValue>
+            <ValidationMessage>Choose a number between 1 and 10000.</ValidationMessage>
+        </max_clients>
+        <ipv4_servers type="BooleanField">
+            <default>1</default>
+            <Required>Y</Required>
+        </ipv4_servers>
+        <ipv6_servers type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </ipv6_servers>
+        <dnscrypt_servers type="BooleanField">
+            <default>1</default>
+            <Required>Y</Required>
+        </dnscrypt_servers>
+        <doh_servers type="BooleanField">
+            <default>1</default>
+            <Required>Y</Required>
+        </doh_servers>
+        <require_dnssec type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </require_dnssec>
+        <require_nolog type="BooleanField">
+            <default>1</default>
+            <Required>Y</Required>
+        </require_nolog>
+        <require_nofilter type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </require_nofilter>
+        <force_tcp type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </force_tcp>
+        <proxy type="TextField">
+            <Required>N</Required>
+        </proxy>
+        <timeout type="IntegerField">
+            <default>2500</default>
+            <Required>Y</Required>
+            <MinimumValue>100</MinimumValue>
+            <MaximumValue>10000</MaximumValue>
+            <ValidationMessage>Choose a number between 100 and 10000.</ValidationMessage>
+        </timeout>
+        <keepalive type="IntegerField">
+            <default>30</default>
+            <Required>Y</Required>
+            <MinimumValue>1</MinimumValue>
+            <MaximumValue>600</MaximumValue>
+            <ValidationMessage>Choose a number between 1 and 600.</ValidationMessage>
+        </keepalive>
+        <cert_refresh_delay type="IntegerField">
+            <default>240</default>
+            <Required>Y</Required>
+            <MinimumValue>1</MinimumValue>
+            <MaximumValue>3600</MaximumValue>
+            <ValidationMessage>Choose a number between 1 and 3600.</ValidationMessage>
+        </cert_refresh_delay>
+        <dnscrypt_ephemeral_keys type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </dnscrypt_ephemeral_keys>
+        <tls_disable_session_tickets type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </tls_disable_session_tickets>
+        <fallback_resolver type="TextField">
+            <default>9.9.9.9:53</default>
+            <Required>Y</Required>
+        </fallback_resolver>
+        <block_ipv6 type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </block_ipv6>
```

Comment:
is this option really needed?

Reply:
I know you are one of the biggest v6 fans :) 
```
## Immediately respond to IPv6-related queries with an empty response
## This makes things faster when there is no IPv6 connectivity, but can
## also cause reliability issues with some stub resolvers.
## Do not enable if you added a validating resolver such as dnsmasq in front
## of the proxy.

```

Reply:
If I understand that correctly, it blocks `AAAA` requests if this option is enabled and there is no IPv6 connectivity on OPNsense itself. This may still break the connections on hosts using a tunnel to get IPv6 connectivity.

Reply:
I'd guess it speeds up DNS when an internal client has v6 enabled but WAN not, if client get's an IP with AAAA it would try this first. With empty response it trys v4 directly.

Reply:
If it cannot resolve AAAA it will try to use A but a client without an IPv6 address should not try to resolve that in such a case because it cannot be reached anyway.

Reply:
I dont know it exactly but since default is 0 it should be OK. I think there is a reason why this parameter is available.